### PR TITLE
[feature](pipeline) avoid unnecessary copying during projection.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes
If a result column id is unique, swap directly.
If there are multiple identical result column id, swap one and copy the rest.

before
```
OLAP_SCAN_OPERATOR  (id=0.  table  name  =  lineitem(lineitem)):
                                    -  BlocksProduced:  sum  194.216K  (194216),  avg  38.843K  (38843),  max  48.653K  (48653),  min  10.724K  (10724)
                                    -  CloseTime:  avg  6.763ms,  max  12.937ms,  min  0ns
                                    -  ExecTime:  avg  1s23ms,  max  1s303ms,  min  277.730ms
                                    -  MemoryUsage:  sum  ,  avg  ,  max  ,  min  
                                        -  PeakMemoryUsage:  sum  0.00  ,  avg  0.00  ,  max  0.00  ,  min  0.00  
                                    -  OpenTime:  avg  4.419ms,  max  6.479ms,  min  2.899ms
                                    -  ProjectionTime:  avg  903.742ms,  max  1s149ms,  min  247.347ms
                                    -  RowsProduced:  sum  787.405897M  (787405897),  avg  157.481179M  (157481179),  max  197.206057M  (197206057),  min  43.521018M  (43521018)
                                    -  RuntimeFilterInfo:  sum  ,  avg  ,  max  ,  min  
                                    -  WaitForDependency[OLAP_SCAN_OPERATOR_DEPENDENCY]Time:  avg  1.76ms,  max  2.945ms,  min  0ns
```

now
```
 OLAP_SCAN_OPERATOR  (id=0.  table  name  =  lineitem(lineitem)):
                                    -  BlocksProduced:  sum  198.75K  (198750),  avg  39.75K  (39750),  max  48.653K  (48653),  min  21.542K  (21542)
                                    -  CloseTime:  avg  8.106ms,  max  15.835ms,  min  0ns
                                    -  ExecTime:  avg  180.862ms,  max  227.702ms,  min  95.371ms
                                    -  MemoryUsage:  sum  ,  avg  ,  max  ,  min  
                                        -  PeakMemoryUsage:  sum  0.00  ,  avg  0.00  ,  max  0.00  ,  min  0.00  
                                    -  OpenTime:  avg  5.692ms,  max  6.603ms,  min  4.609ms
                                    -  ProjectionTime:  avg  52.289ms,  max  64.535ms,  min  30.375ms
                                    -  RowsProduced:  sum  805.805177M  (805805177),  avg  161.161035M  (161161035),  max  197.206057M  (197206057),  min  87.392577M  (87392577)
                                    -  RuntimeFilterInfo:  sum  ,  avg  ,  max  ,  min  
                                    -  WaitForDependency[OLAP_SCAN_OPERATOR_DEPENDENCY]Time:
```


For this type of projection with many references, there is a significant performance improvement.
Because it avoids creating copies.

```
 0:VOlapScanNode(258)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|      TABLE: tpch_sf100.lineitem(lineitem), PREAGGREGATION: ON                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
|      PREDICATES: (l_shipdate[#0] <= '1998-09-02')                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
|      partitions=1/1 (lineitem), tablets=768/768, tabletList=866164,866166,866168 ...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
|      cardinality=600037902, avgRowSize=0.0, numNodes=1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|      pushAggOp=NONE                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|      final projections: l_returnflag[#9], l_linestatus[#10], l_quantity[#5], l_extendedprice[#6], l_discount[#7], l_tax[#8]                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
|      final project output tuple id: 1         
```

<!--Describe your changes.-->

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/core/block.cpp [hunk 1]
```diff
@@ -686,7 +688,7 @@ void Block::clear() {
 std::string Block::print_use_count() {
     std::stringstream ss;
```
Comment (HappenLee) [inline be/src/vec/core/block.cpp:689]: do refactor use fmt lib

File: be/src/vec/core/block.cpp [hunk 2]
```diff
@@ -728,6 +730,87 @@ void Block::shuffle_columns(const std::vector<int>& result_column_ids) {
...
+                } else {
+                    MutableColumnPtr mut_to = (*std::move(to)).mutate();
+                    mut_to->insert_range_from(*from, 0, rows);
+                    to = std::move(mut_to);
+                }
+            }
+        };
+        if (is_column_const(*origin_column)) {
+            auto column_ptr = origin_column->convert_to_full_column_if_const();
+            insert_column(to, column_ptr, rows, false);
```
Comment (HappenLee) [inline be/src/vec/core/block.cpp:766]: the column_ptr is temporary mem，so here better `use_swap` = true

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Pull request closed without merging**